OrbitGl/App.cpp

```diff
@@ -1152,11 +1152,21 @@ void OrbitApp::InitializeClientTransactions() {
   frame_pointer_validator_client_ =
       std::make_unique<FramePointerValidatorClient>(this,
                                                     transaction_client_.get());
-  process_memory_client_ =
-      std::make_unique<ProcessMemoryClient>(transaction_client_.get());
 }
 
 //-----------------------------------------------------------------------------
 void OrbitApp::FilterFunctions(const std::string& filter) {
   m_LiveFunctionsDataView->OnFilter(filter);
 }
+
+void OrbitApp::GetRemoteMemory(uint32_t pid, uint64_t address, uint64_t size,
+                               const ProcessMemoryCallback& callback) {
+  thread_pool_->Schedule([this, pid, address, size, callback] {
+    std::string memory = process_manager_->GetProcessMemory(pid, address, size);
+    main_thread_executor_->Schedule([memory = std::move(memory), callback] {
+      // TODO: This makes a copy.
+      std::vector<uint8_t> data{memory.begin(), memory.end()};
```

Comment:
I'd like your opinion on this copy. Should I leave it for now, should I change `ProcessMemoryCallback` and the other affected functions/methods to also accept a `std::string`, or something else? Using `bytes` in `GetProcessMemoryResponse` causes a `std::string` and I don't know if there's a better way to obtain a `std::vector<uint8_t>`.

Reply:
I would say pass std::string and cast .data() to appropriate time when it is consumed.

Reply:
So this is done.